framework/src/main/java/org/checkerframework/framework/util/ContractsUtils.java

```diff
@@ -483,23 +526,85 @@ private AnnotationMirror getAnnotationMirrorOfMetaAnnotation(
             }
         }
 
-        // Check type-system-specific annotations.
+        // Check for type-system-specific annotations.
         List<Pair<AnnotationMirror, AnnotationMirror>> declAnnotations =
                 factory.getDeclAnnotationWithMetaAnnotation(
                         methodElement, ConditionalPostconditionAnnotation.class);
         for (Pair<AnnotationMirror, AnnotationMirror> r : declAnnotations) {
             AnnotationMirror anno = r.first;
             AnnotationMirror metaAnno = r.second;
-            AnnotationMirror postcondAnno = getAnnotationMirrorOfMetaAnnotation(metaAnno, anno);
-            if (postcondAnno == null) {
-                continue;
-            }
-            List<String> expressions =
-                    AnnotationUtils.getElementValueArray(anno, "expression", String.class, false);
-            boolean annoResult =
-                    AnnotationUtils.getElementValue(anno, "result", Boolean.class, false);
-            for (String expr : expressions) {
-                result.add(new ConditionalPostcondition(expr, annoResult, postcondAnno, anno));
+
+            result.addAll(getConditionalPostConditionAnnotations(anno, metaAnno));
+        }
+
+        return result;
+    }
+
+    /**
+     * Returns a collection of conditional postcondition annotations that can consist of
+     * {@code @EnsuresNonNullIf},{@code @EnsuresKeyForIf},{@code @EnsuresLTLengthOfIf},
+     * {@code @EnsuresLockHeldIf}, {@code @EnsuresMinLenIf} or any other conditional postcondition
+     * annotation developed in the future.
+     *
+     * @param anno wrapper annotation if multiple annotations are present, otherwise single
+     *     annotation
+     * @param @param metaAnno conditional postcondition meta-annotation that indicates that an
+     *     annotation is a conditional postcondition annotation
+     */
+    private Set<ConditionalPostcondition> getConditionalPostConditionAnnotations(
+            AnnotationMirror anno, AnnotationMirror metaAnno) {
+
+        Set<ConditionalPostcondition> result = new LinkedHashSet<>();
+
+        Map<? extends ExecutableElement, ? extends AnnotationValue> valmap;
```

Comment:
This variable name communicates nothing about what the variable represents.  Use a better name, and perhaps add a comment.

Reply:
Channged the name to `annovalue` and added a comment.